pkgs/development/libraries/folly/default.nix

```
@@ -26,13 +26,13 @@
 
 stdenv.mkDerivation rec {
```

```suggestion
let
  stdenv' = if stdenv.isDarwin && stdenv.isx86_64
    then stdenv.override (oldStdenv: {
      buildPlatform = oldStdenv.buildPlatform // { darwinMinVersion = "10.13"; };
      targetPlatform = oldStdenv.targetPlatform // { darwinMinVersion = "10.13"; };
      hostPlatform = oldStdenv.hostPlatform // { darwinMinVersion = "10.13"; };
    })
    else stdenv;
in stdenv'.mkDerivation rec {
```

This gem from @reckenrode in #253031 appears to address my build failure.

---

Instead of doing the override manually, you can use `overrideSDK` to change the deployment target. You’ll need to add `overrideSDK` to the derivation’s arguments.

```suggestion
let
  stdenv' =
    if stdenv.isDarwin && stdenv.isx86_64 then
      overrideSDK stdenv { darwinMinVersion = "10.13"; }
    else
      stdenv;
in
stdenv'.mkDerivation rec {
```

---

Ah I just pushed the above suggestion. I don't have a darwin system to test, so please let know whichever works.

---

I think they all work, but the cmake flag is the most straightforward (and I tested it)

---

Switched to the cmake flag.

@risicle can you check if I got the changes right?